### PR TITLE
BROC-205: environment slug always showing as updating in tf plan

### PR DESCRIPTION
**Pull request by JureMalovrh**
We were updating env slugs on the fly - this causes issues when terraform tries to re-calcualate the plan for change.
There is no clear or nice way to remove the plan because SDK calls DiffSuppressFunc on individual values not on whole plan as one.

Current resolution is to introduce id field that is composed of our calculated slug and leave slug as-is to avoid un-necessary changes.

To test:
- Go to master & install provider
- apply `sleuth_code_change_source` with `environment_mapping`
- after successful apply - try to plan/apply again - see changes in `env_mapping`
- switch to this branch and install provider
- try to plan/apply - observe no changes in `env_mapping`

Resolves: https://github.com/sleuth-io/terraform-provider-sleuth/issues/58

### CODE REVIEW

**Comment by JureMalovrh** [comment]
@mrdon let's discuss this when you return

File: internal/provider/resource_code_change_source.go [hunk 1]
```diff
@@ -76,8 +76,9 @@ func resourceCodeChangeSource() *schema.Resource {
 				},
 			},
 			"environment_mappings": {
-				Type:     schema.TypeList,
-				Required: true,
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Environment mappings of the environment. They must be ordered by environment_slug ascending to avoid Terraform plan changes.",
```
Comment (JureMalovrh) [inline internal/provider/resource_code_change_source.go:81]: @adamchatko could you please write/review this in a more coherent way :)

**Comment by mrdon** [comment]
Didn't run but seems fine. My main request is to make sure it doesn't break existing customers

**Review by mrdon — APPROVED** [review]

**Comment by jrgarcia** [comment]
Running locally, I'm getting this

```
Terraform will perform the following actions:

  # sleuth_code_change_source.cd will be updated in-place
  ~ resource "sleuth_code_change_source" "cd" {
        id                   = "testing/foo"
        name                 = "foo"
        # (6 unchanged attributes hidden)

      ~ environment_mappings {
          ~ environment_slug = "production" -> "staging"
            id               = "testing/production"
            # (1 unchanged attribute hidden)
        }
      ~ environment_mappings {
          ~ environment_slug = "staging" -> "production"
            id               = "testing/staging"
            # (1 unchanged attribute hidden)
        }

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

**Comment by jrgarcia** [comment]
Ah, I see that this is do to the `environment_mappings` in the example file not being ordered alphabetically. I swapped and see it working as expected. That isn't exactly ideal though. I'll take a look, but if you have any ideas what my prevent us from needing to have them ordered a certain way, I'd say we should do that. That's a pretty significant "gotcha" to have to consider.

**Comment by JureMalovrh** [comment]
> 



> Ah, I see that this is due to the `environment_mappings` in the example file not being ordered alphabetically. I swapped and see it working as expected. That isn't exactly ideal though. I'll take a look, but if you have any ideas what my prevent us from needing to have them ordered a certain way, I'd say we should do that. That's a pretty significant "gotcha" to have to consider.

Yeah it's not a nice solution but currently, with SDK hacks for nested objects, it is sadly the only one we can do. I have added it in the docs that this should be ordered but yeah the issue is that our API automatically sorts it alphabetically which then confuses Terraform. We don't have any other ordering property in our API so we have to comply 😢

**Review by jrgarcia — APPROVED** [review]

**Pull request merged**